_posts/2019-08-05-cognito-azuread-sso.markdown

```diff
@@ -0,0 +1,91 @@
+---
+layout: post
+title:  "Securing your web applications in AWS using SSO with Azure Active Directory"
+date:   2019-08-05
+tags: aws cognito oidc saml2 sso azure
+author: Paul Symons
+image: 
+---
+
+It's a fairly common scenario where you want to secure a web application in the cloud; but with so many technologies and different ways to secure your applications, it can be difficult to know where to get started. You may have questions, such as:
+
+- should I handle authentication within my web application itself, or within infrastructure?
+- how can I integrate with my organisation's identity provider?
+- can I automate the securing of my web application?
+- how can others in my organisation leverage this pattern?
+
+At [Mechanical Rock](https://www.mechanicalrock.io/our-expertise), two of our pillars of excellence are
+* [Cloud Native Applications](https://mechanicalrock.io/our-expertise/cloud-native-applications), and 
+* [Enterprise Devops](https://mechanicalrock.io/our-expertise/enterprise-devops)
+
+ We leverage _best of breed_ cloud native and managed services to build high quality, cutting edge solutions. Delivering these solutions using Infrastructure-As-Code (IAC) is not simply part of our commitment to architecting steadfast **continuous integration delivery** workflows; it underpins the way we collaborate and share learnings with the client teams we engage with.
+
+### An opinionated web authentication method
+
+Most large organisations invest in single sign on capabilities for their colleagues, which is typically good for security practice - provide one authentication method, and make it strong, with password rotations and multi-factor authentication.
+
+The solution we will focus on here is a pattern we commonly use when working with large enterprise clients, that have an established identity provider:
+
+- Microsoft Azure Active Directory as _Identity Provider_
+- AWS Cognito as _Authentication Service_
+- AWS Application Load Balancer as authentication proxy to our web application
+
+![cognito sso webapp design](/img/cognito-sso-design.jpg)
+
+Our example assumes a web application running on ECS or EC2 or similar, but in reality it can be anything that can update a Load Balancer Target Group.
```

Comment:
Not serverless?

Reply:
i say no, if you are going serverless you are better with a SPA auth solution or using cognito user pool with API gateway.